records_mover/records/validated_records_hints.py

```diff
@@ -10,6 +12,49 @@
                     VALID_DATETIMEFORMATTZS, VALID_DATETIMEFORMATS)
 from .hints import cant_handle_hint
 
+HintT = TypeVar('HintT')
+
+
+class HintValidator:
+    def __init__(self,
+                 fail_if_cant_handle_hint: bool,
+                 hints: RecordsHints):
+        self.fail_if_cant_handle_hint = fail_if_cant_handle_hint
+        self.hints = hints
+
+        class Hint(Generic[HintT]):
+            def __init__(self,
+                         type_: Type[HintT],
+                         hint_name: str,  # TODO: HintName,
+                         default: HintT):
+                self.default = default
+                self.type_ = type_
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker